http-clients/src/main/java/com/palantir/remoting/http/FeignClientFactory.java

```diff
@@ -117,68 +135,129 @@ public static FeignClientFactory of(
-        return createProxy(sslSocketFactory, uri, type, options);
+    public <T> T createProxy(Optional<SSLSocketFactory> sslSocketFactory, String uri, Class<T> type,
+            ProxyConfigurationProvider proxyConfigurationProvider) {
+        return createProxy(sslSocketFactory, uri, type, proxyConfigurationProvider, options);
+    }
+
+    /**
+     * Constructs a dynamic proxy for the specified type, using the supplied SSL factory if is present, and {@link
+     * okhttp3.OkHttpClient} wrapped in {@link feign.okhttp.OkHttpClient}.
+     */
+    public <T> T createProxy(Optional<SSLSocketFactory> sslSocketFactory, String uri, Class<T> type,
+            ProxyConfigurationProvider proxyConfigurationProvider, Request.Options requestOptions) {
+        return createProxy(sslSocketFactory, ImmutableSet.of(uri), type, proxyConfigurationProvider, requestOptions);
     }
 
+    /**
+     * Constructs a dynamic proxy for the specified type, using the supplied SSL factory if is present, and
+     * {@link okhttp3.OkHttpClient} wrapped in {@link feign.okhttp.OkHttpClient}. A {@link FailoverFeignTarget}
+     * is used to cycle through the given uris on failure if there's more than one uri.
+     */
+    public <T> T createProxy(Optional<SSLSocketFactory> sslSocketFactory, Set<String> uris, Class<T> type) {
+        return createProxy(sslSocketFactory, uris, type, options);
+    }
 
     /**
-     * Constructs a dynamic proxy for the specified type, using the supplied SSL factory if is present, and feign {@link
-     * feign.Client.Default} HTTP client. A {@link FailoverFeignTarget} is used to cycle through the given uris on
-     * failure.
+     * Constructs a dynamic proxy for the specified type, using the supplied SSL factory if is present, and {@link
+     * okhttp3.OkHttpClient} wrapped in {@link feign.okhttp.OkHttpClient}. A {@link FailoverFeignTarget} is used
+     * to cycle through the given uris on failure if there's more than one uri.
      */
     public <T> T createProxy(Optional<SSLSocketFactory> sslSocketFactory, Set<String> uris, Class<T> type,
             Request.Options requestOptions) {
-        FailoverFeignTarget<T> target = new FailoverFeignTarget<>(uris, type, backoffStrategy);
-        Client client = clientSupplier.createClient(sslSocketFactory, userAgent);
-        client = target.wrapClient(client);
-        return Feign.builder()
+        return createProxy(sslSocketFactory, uris, type, defaultProxyConfigurationProvider, requestOptions);
+    }
+
+    /**
+     * Constructs a dynamic proxy for the specified type, using the supplied SSL factory if is present, proxy
+     * configuration provider, and {@link okhttp3.OkHttpClient} wrapped in {@link feign.okhttp.OkHttpClient}.
+     * A {@link FailoverFeignTarget} is used to cycle through the given uris on failure if there's more than one uri.
+     */
+    public <T> T createProxy(Optional<SSLSocketFactory> sslSocketFactory, Set<String> uris, Class<T> type,
```

Comment:
Rob and I have been discussing at least hiding Feign from the API in this regardless. 

Conjure gives us optionality for change, but isn't a golden ticket. Certainly things are simpler if we hide the impl details, though.